src/Microsoft.SqlTools.Migration/Contracts/ValidateNetworkFileShare.cs

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.Migration.Contracts
+{
+    public class ValidateNetworkFileShareRequestParams
+    {
+        public string Path { get; set; }
+        public string Username { get; set; }
+        public string Password { get; set; }
+    }
```

Comment:
Should passwords/credentials be SecureString?

Reply:
Given that this is already doing that in the existing code changing it now seems out of scope of this PR

https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Migration/Contracts/ValidateNetworkFileShare.cs#L14

So I'd suggest filing an issue for your team to follow up with after if you think it's a concern - although my understanding is that given that accessing process memory is going to require elevated privileges anyways (at which point you're already compromised) that it isn't a security concern to have it in plaintext in places like this. Especially since this isn't being stored long term anywhere (I hope) - but rather just a transient thing that's only around for the lifetime of the request. 

@shueybubbles You also mentioned something about SecureString being not so great anyways, do you happen to have a link to where you saw that from?

Reply:
in the horse's mouth!

https://learn.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-7.0#remarks